src/plugins/template/src/config.cpp

```diff
@@ -41,7 +41,7 @@ Configuration::Configuration(const ov::AnyMap& config, const Configuration& defa
         } else if (ov::inference_num_threads.name() == key) {
             int val;
             try {
-                val = value.as<int>();
+                val = value.as<decltype(ov::inference_num_threads)::value_type>();
```

Comment:
I think it should not be required as numerical values can be converted to other numerical type.